tests/ServerTest.php

```diff
@@ -1023,6 +1023,162 @@ function ($data) use (&$buffer) {
         $this->assertInstanceOf('InvalidArgumentException', $error);
     }
 
+    public function testInvalidChunkHeaderResultsInErrorResponse()
+    {
+        $error = null;
+        $server = new Server($this->socket);
+        $server->on('request', function ($request, $response) use (&$error){
+            $request->on('error', function ($ex) use (&$error) {
+                $error = $ex;
+            });
```

Comment:
I think the updated tests are easier to grasp :+1: *IF* the exception class is really important here (I'm undecided about this), then you may want to use `expectCallableOnceWith(…)` and family?